_posts/2018-08-09-deploying-esxi-through-foreman.md

```diff
@@ -0,0 +1,402 @@
+      </p>
+    * Click Submit
+
+  * **1.3.2:** Create the Kickstart Template
+    * Click on **Create Template**
+    * Name - ESXi Minimal Kickstart
+    * A sample template:  
+      ```
+      vmaccepteula
+      keyboard 'US Default'
+      reboot
+      rootpw P@ssw0rd
+      install --firstdisk --overwritevmfs --novmfsondisk
+
+      # Set the network to DHCP on the first network adapter
+      network --bootproto=dhcp --device=<%= @host.mac %>
+
+      %post --interpreter=busybox
+      # Add temporary DNS resolution so the foreman call works
+      echo "nameserver <%= @host.subnet.dns_primary %>" >> /etc/resolv.conf
+
+      # Inform Foreman that we are done.
+      wget -O /dev/null <%= foreman_url('built') %>
+      echo "Done with Foreman call"
+      ```
+
+    * Template Type - **Provisioning template**
+      <p float='left' align='left'>
+        <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/esxi_ks_template_new_type.png" width="400" />
+      </p>
+    * Template Association - Select **"ESXi 6.7 Build 8169922"**
+    * Click Submit
+
+**1.4: Set default provisioning templates**  
+  Now that the templates are created, we need to set those as default.
+  * Navigate to https://{foreman-url}/operatingsystems and click on the newly created ESXi OS **"ESXi 6.7 Build 8169922"**
+  * Open the "Templates" tab
+  * Select the newly created provisioning templates from the drop down boxes and press Submit.
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/os_default_templates.png" width="400" />
+    </p>
+
+### 2: ESXi installation in Legacy BIOS mode
+
+---
+
+The ESXi PXELinux mboot.c32 module only supports syslinux bootloader (pxelinux.0) version 3.86. You may replace the newer pxelinux.0 file that comes with the default Foreman installation with pxelinux.0 v3.86 but then you might face issues with newer OSes. So we will use a different method.
+
+**2.1: Prepare the bootloader**  
+On the Foreman server, extract syslinux-3.86.tar.gz. Copy `syslinux-3.86/core/pxelinux.0` to the tftp root directory (typically `/var/lib/tftpboot`) and rename it to `pxelinux-3.86.0`
+
+**2.2: Create the bootloader entry in Foreman**  
+Now comes the interesting part. We will create a new PXELoader entry in Foreman for `pxelinux-3.86.0`. Huge thanks to Lukáš Zapletal for [pointing this out](https://community.theforeman.org/t/edit-dhcpd-leases-through-a-foreman-hook-script/9957/2).  
+  * **2.2.1:** Edit the file -   `/usr/share/foreman/app/models/concerns/pxe_loader_support.rb`
+    <pre><code>
+    def all_loaders_map(precision = 'x64')
+      {
+        "None" => "",
+        "PXELinux BIOS" => "pxelinux.0",
+        "PXELinux UEFI" => "pxelinux.efi",
+        <b>"ESXi PXELinux BIOS" => "pxelinux-3.86.0",</b> # Add this line
+        ...
+    </pre></code>
```

Comment:
Feel free to file a PR to add this line. I'd not mention ESXi in the loader name tho to prevent user confusion.

Reply:
What would you like to call it? "PXELinux v3.86"?

Reply:
Also what about the EFI bootloader entry? "ESXi mboot UEFI"? Do you want that changed as well?

Reply:
I'd suggest naming this "PXELinux Alt BIOS" -> `pxelinux.alt/pxelinux.0` because with symlink you could avoid the incompatibility with modules.